students/Cracolici_Jon/lesson04/JonCracolici_L4_Mailroom2.py

```diff
@@ -0,0 +1,171 @@
+    jeff_bezos = {'fname': 'Jeff_Bezos', 'donor': 'Jeff Bezos', 'totaldonated': 877.33, 'numdonations': 1, 'avgdonated': 877.33, 'lastdonation': 877.33}
+    paul_allen = {'fname': 'Paul_Allen', 'donor': 'Paul Allen', 'totaldonated': 708.42, 'numdonations': 3, 'avgdonated': 236.14, 'lastdonation': 100.95}
+    
+    dB = {'william_gates_iii': william_gates_iii,'mark_zuckerberg': mark_zuckerberg,'jeff_bezos': jeff_bezos, 'paul_allen': paul_allen}
+    return dB
+#Functions relating to the start menu
+def start_control(dB):
+    """Creates the control flow for the start menu."""
+    start_menu_display()
+    response = start_user_input()
+    return startmenulogic[response](dB)
+def start_menu_display():
+    """Prints the start menu to the screen."""
+    print('Welcome to our mailroom app!')
+    print('Please select the number of your choice from the following options')
+    print('1) Send a Thank You')
+    print('2) Create a Report')
+    print('3) Send Letters to Everyone')
+    print('4) Quit')
+def start_user_input():
+    """Collects user input. I separate this for easier error handling in future."""
+    task = input('Please type the number of your selection.')
+    return task
+#Functions relating to the Send a Thank You Menu
+def saty_control(dB):
+    """Creates the control flow for the send a thank you menu."""
+    saty_menu_display()
+    response = saty_user_input()
+    return satymenulogic[response](dB)
+def saty_menu_display():
+    """Prints the send a thank you menu to the screen."""
+    print('You have chosen to send a thank you note to a donor')
+    print('Please select the number of you choice from the following options')
+    print('1) See a list of current donors')
+    print('2) Send a thank you to a current donor for a new gift')
+    print('3) Send a thank you to a first time donor')
+    print('4) Return to the start menu')
+def saty_user_input():
+    """Collects user input. I separate this for easier error handling in future."""
+    task = input('Please type the number of your selection.')
+    return task
+def nd_control(dB):
+    """Function that calls functions to create and add a new donor to database, print letter to screen."""
+    newdonordict = donor_input(dB)
+    if newdonordict == False:
+        return False
+    #will add try except to check for donor actually being new in L5
+    new_donor(newdonordict, dB)
+    note_gen(dB[newdonordict['donornamekey']])
+    return dB#, start_control(dB)
+def ud_control(dB):
+    """Function that calls functions to update a donor in the database, print letter to screen."""
+    updonordict = donor_input(dB)
+    if updonordict == False:
+        return False
+    #will add try except to check for donor actually being new in L5
+    update_donor(updonordict, dB)
+    note_gen(dB[updonordict['donornamekey']])
+    return dB#, start_control(dB)
```

Comment:
Note that if the user selects an existing donor, but doesn't input the correct name the program breaks.  Suggest an exception for such cases.